pkg/cloudprovider/aws/factory.go

```diff
@@ -53,7 +54,11 @@ type Factory struct {
 }
 
 func NewFactory(options cloudprovider.Options) *Factory {
-	sess := withUserAgent(withRegion(session.Must(session.NewSession(&aws.Config{STSRegionalEndpoint: endpoints.RegionalSTSEndpoint}))))
+	sess := withUserAgent(withRegion(session.Must(
+		session.NewSession(request.WithRetryer(
+			&aws.Config{STSRegionalEndpoint: endpoints.RegionalSTSEndpoint},
+			// use a custom retryer to take care of ec2 eventual consistency
```

Comment:
Awesome. I think we can skip this comment, though.

Reply:
👍